### PR TITLE
[TRA-220] add activation threshold on vaults

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
do not activate a vault if it has no perp position and has less than some threshold of quote quantums

default value is 1000 USDC

### Test Plan
unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new parameter to specify the minimum quote asset amount required for vault activation without perpetual positions.
	- Added a new reason for order removal related to isolated subaccount constraints in the API documentation.

- **Bug Fixes**
	- Implemented checks to skip processing for vaults not meeting the new activation threshold.

- **Documentation**
	- Updated Swagger UI documentation to reflect new enum values and descriptions.

- **Tests**
	- Enhanced test coverage to include scenarios with the new activation threshold parameter.

- **Refactor**
	- Updated various parts of the code to integrate the new activation threshold logic.

- **Chores**
	- Added necessary imports and updated default parameters to support the new features.

### CODE REVIEW

**Review by Christopher-Li — APPROVED** [review]

File: proto/dydxprotocol/vault/params.proto [hunk 1]
```diff
@@ -25,4 +27,14 @@ message Params {
 
   // The duration that a vault's orders are valid for.
   uint32 order_expiration_seconds = 6;
+
+  // The number of quote quantums in quote asset that a vault with no perpetual
+  // positions must have to activate, i.e. if a vault has no perpetual positions
+  // and has strictly less than this amount of quote asset, it will not
+  // activate.
+  bytes activation_threshold_quote_quantums = 7 [
```
Comment (teddyding) [inline proto/dydxprotocol/vault/params.proto:35]: Nit: consider using uint64 why is sufficient for all realistic threshold here (`uint64` max = 18 trillion USDC)

Comment (teddyding) [inline proto/dydxprotocol/vault/params.proto:35]: Did we ever create upgrade handler for `x/vault`? If so do we need to initialize this value as well?

File: protocol/x/vault/keeper/orders.go [hunk 2]
```diff
@@ -34,6 +36,22 @@ func (k Keeper) RefreshAllVaultOrders(ctx sdk.Context) {
 			continue
 		}
 
+		// Skip if vault has no perpetual positions and strictly less than `activation_threshold_quote_quantums` USDC.
+		params := k.GetParams(ctx)
```
Comment (teddyding) [inline protocol/x/vault/keeper/orders.go:40]: We don't need to get this params for each vaults

File: protocol/x/vault/keeper/orders.go [hunk 3]
```diff
@@ -34,6 +36,22 @@ func (k Keeper) RefreshAllVaultOrders(ctx sdk.Context) {
...
+		vault := k.subaccountsKeeper.GetSubaccount(ctx, *vaultId.ToSubaccountId())
+		if vault.PerpetualPositions == nil || len(vault.PerpetualPositions) == 0 {
+			usdcQuoteQuantums := dtypes.NewInt(0)
```
Comment (teddyding) [inline protocol/x/vault/keeper/orders.go:43]: Use `subaccount.GetUsdcPosition()` instead

File: proto/dydxprotocol/vault/params.proto (hunk 1 above)
Comment (tqin7) [reply proto/dydxprotocol/vault/params.proto:35]: i agreed that uint64 is probably sufficient but a deposit to vault is represented as a serializable int already and wanted to keep it consistent

Comment (tqin7) [reply proto/dydxprotocol/vault/params.proto:35]: yep upgrade handler is created for `x/vault`. this value should be automatically initialized as the upgrade handler [calls `DefaultParams()` to get params to set](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/app/upgrades/v5.0.0/upgrade.go#L310)

**Comment by tqin7** [comment]
@mergifyio backport release/protocol/v5.x

**Pull request merged**